Fix installation on Amazon Linux and add support for Ubuntu 15.04

tenyo opened the pull request:
This addresses #15 by fixing telegraf installation on Amazon Linux.

Will also use RHEL yum repo (instead of CentOS) if the server is running RHEL.

cgatesman commented:
:+1: 

Do you need to update the CHANGELOG.md, and perhaps README.md?

fishnix commented:
@cgatesman good call (+ metadata.rb), but for the public cookbooks we are going to do that on master (or the long running v0.10 dev branch in this case which will get merged into master soon), prior to releasing the cookbook to the supermarket.  This way we could potentially collect a few fixes in a single release.

@tenyo thanks for doing this, the only issue is there's not an easy way to do integration tests on them.  We should however add them as tested platforms in the unit tests (looks like there's support https://github.com/customink/fauxhai/tree/v3.0.1/lib/fauxhai/platforms).  We should also consider adding support for `Ubuntu Server 15.10`.

cgatesman commented:
> @cgatesman good call (+ metadata.rb), but for the public cookbooks we are going to do that on master (or the long running v0.10 dev branch in this case which will get merged into master soon), prior to releasing the cookbook to the supermarket.  This way we could potentially collect a few fixes in a single release.

Yeah.  I figured you could maintain those changes on the v0.10 branch, instead of trying to figure out what all changed and writing it all down at once at the merge.

tenyo commented:
@fishnix thanks, I'll add it to chefspec

jwinfield commented:
:+1:

tenyo commented:
Added rspec test for Amazon Linux. No way to test with kitchen but I verified it actually works on Amazon Linux.
Added rspec and kitchen test for Ubuntu 15.04 (15.10 box not available yet).
Bumped the default telegraf version to 0.10.3.

fishnix commented:
looks good :+1:
